store/activities.js

```diff
@@ -2,31 +2,45 @@ const config = [
   {
     id: 'overview',
     path: 'overview',
+    icon: 'dashboard',
   },
   {
     id: 'content',
     path: 'content',
+    icon: 'document',
   },
   {
     id: 'depositDates',
     path: 'deposit-dates',
+    icon: 'calendar-check',
   },
   {
     id: 'plugins',
     path: 'plugins',
-  },
-  {
-    id: 'plugins/discovery',
-    path: 'plugins/discovery',
-  },
-  {
-    id: 'plugins/recommender',
-    path: 'plugins/recommender',
+    icon: 'puzzle',
+    subPaths: [
+      {
+        id: 'plugins/discovery',
+        path: 'plugins/discovery',
+      },
+      {
+        id: 'plugins/recommender',
+        path: 'plugins/recommender',
+      },
+    ],
   },
 ]
 
 const configMap = new Map(config.map(item => [item.id, item]))
 
 config.get = (...args) => configMap.get(...args)
+config.find = (...args) =>
+  config
+    .reduce((acc, curr) => {
+      acc.push(curr)
+      if (curr.subPaths) acc.push(...curr.subPaths)
```

Comment:
Technically it's not correct. More specific routes should go first. In our case let's imagine that you match regular expression: `/plugins` would match `/plugins/discovery` so you always would get `plugins`.

```suggestion
      acc.push(...(curr.children ?? []), curr)
```

Reply:
Sorry, I don't get your example. If you use the regular expression the find method return you first match, which is `/plugins` and that's correct.

Reply:
Ah, I get it

Reply:
but anyway why would I use `/plugins` for matching /plugins/discovery?

Reply:
It's not about using plugins for matching `/plugins/discovery`. It's basically assumption that URL can be `/plugins/discovery#custom-settings` or `/plugins?viewAs=-1`. We should check only start of the URL and in this case the first option, i.e. `/plugins` would match.

Reply:
And that's correct, isn't it? So if I have `/plugins/discovery#custom-settings`, `/plugins/recommender#custom-settings` and `/plugins#custom-settings` and I use only start url, i.e. `/plugins` which path should be matched? From what you are saying it should match `/plugins/discovery` and how do you know it's better then `/plugins/recommender`?

Reply:
When you have a `test` and an actual `location`, and your tests are ordered
```js
[
  /^\/plugins/,
  /^\/plugins\/discovery/,
]
```

then testing a `/plugins/discovery#custom-settings` would match the first one, i.e. `/^\/plugins/` what is not correct.

If you use not `RegExp.prototype.test` but the `String.prototype.startsWith` the result would be the same.

---

Also, I am saying that we should rely on `startsWith` because there may be parameters in the end of the URL.